core/src/main/java/com/rudderstack/android/sdk/core/DBPersistentManager.java

```diff
@@ -386,31 +380,26 @@ private DBPersistentManager(Application application) {
        save them to the sqlite db sequentially.
      */
     void startHandlerThread() {
-        // Need to perform db operations on a separate thread to support strict mode.
-        ExecutorService executor = Executors.newSingleThreadExecutor();
-
-        Runnable runnable = new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    synchronized (DBPersistentManager.QUEUE_LOCK) {
-                        SQLiteDatabase database = DBPersistentManager.this.getWritableDatabase();
-                        dbInsertionHandlerThread = new DBInsertionHandlerThread("db_insertion_thread", database);
-                        dbInsertionHandlerThread.start();
-                        for (Message msg : queue) {
-                            addMessageToHandlerThread(msg);
-                        }
+        Runnable runnable = () -> {
+            try {
+                synchronized (DBPersistentManager.QUEUE_LOCK) {
+                    SQLiteDatabase database = DBPersistentManager.this.getWritableDatabase();
+                    dbInsertionHandlerThread = new DBInsertionHandlerThread("db_insertion_thread", database);
+                    dbInsertionHandlerThread.start();
+                    for (Message msg : queue) {
+                        addMessageToHandlerThread(msg);
                     }
-                } catch (SQLiteDatabaseCorruptException ex) {
-                    RudderLogger.logError(ex);
-                } catch (ConcurrentModificationException ex) {
-                    RudderLogger.logError(ex);
-                } catch (NullPointerException ex) {
-                    RudderLogger.logError(ex);
                 }
+            } catch (SQLiteDatabaseCorruptException ex) {
+                RudderLogger.logError(ex);
+            } catch (ConcurrentModificationException ex) {
+                RudderLogger.logError(ex);
+            } catch (NullPointerException ex) {
+                RudderLogger.logError(ex);
             }
         };
-        Future future = executor.submit(runnable);
+        // Need to perform db operations on a separate thread to support strict mode.
+        Future<?> future = Executors.newSingleThreadExecutor().submit(runnable);
         try {
             // todo: shall we add some timeout here ?
             future.get();
```

Comment:
since we are not utilising the Future reponse, we might use "executor.execute()"

Reply:
yes, we are not using the future's response but we do need to await until the runnable finishes its execution, hence i went with using submit and getting future.

Reply:
using executor.execute() now